src/lib/source-handlers/bitbucket-server/project-is-empty.ts

```diff
@@ -0,0 +1,24 @@
+import { getBitbucketServerToken } from './get-bitbucket-server-token';
+import { fetchAllRepos } from './list-repos';
+import { BitbucketServerRepoData } from './types';
+
+export async function bitbucketServerProjectIsEmpty(
+  projectKey: string,
+  sourceUrl?: string,
+): Promise<boolean> {
+  const bitbucketServerToken = getBitbucketServerToken();
+  if (!sourceUrl) {
+    throw new Error(
+      'Please provide required `sourceUrl` for Bitbucket Server source',
+    );
+  }
+  const repos = (await fetchAllRepos(
+    sourceUrl,
+    projectKey,
+    bitbucketServerToken,
+  )) as BitbucketServerRepoData[];
+  if (!repos || repos.length === 0) {
```

Comment:
Is `return !repos || repos.length === 0` not the same?

Reply:
Not sure, in previous cases the !repos wasn't enough since we got an empty array and that failed to pass the "!repos" condition.

Reply:
done